contracts/RouterStrategy.sol

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: AGPL-3.0
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import {BaseStrategy} from "@yearnvaults/contracts/BaseStrategy.sol";
+import {
+    SafeERC20,
+    SafeMath,
+    IERC20,
+    Address
+} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+interface IVault is IERC20 {
+    function token() external view returns (address);
+
+    function decimals() external view returns (uint256);
+
+    function deposit() external;
+
+    function pricePerShare() external view returns (uint256);
+
+    function withdraw(
+        uint256 amount,
+        address account,
+        uint256 maxLoss
+    ) external returns (uint256);
+}
+
+contract RouterStrategy is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    string internal strategyName;
+    IVault public yVault;
+    uint256 public maxLoss;
+
+    constructor(
+        address _vault,
+        address _yVault,
+        string memory _strategyName
+    ) public BaseStrategy(_vault) {
+        _initializeThis(_yVault, _strategyName);
+    }
+
+    function _initializeThis(address _yVault, string memory _strategyName)
+        internal
+    {
+        yVault = IVault(_yVault);
+        strategyName = _strategyName;
+    }
+
+    function name() external view override returns (string memory) {
+        return strategyName;
+    }
+
+    function estimatedTotalAssets() public view override returns (uint256) {
+        return balanceOfWant().add(valueOfInvestment());
+    }
+
+    function delegatedAssets() external view override returns (uint256) {
+        return vault.strategies(address(this)).totalDebt;
+    }
+
+    function prepareReturn(uint256 _debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 _profit,
+            uint256 _loss,
+            uint256 _debtPayment
+        )
+    {
+        uint256 balanceInit = balanceOfWant();
+        _takeVaultProfit();
+        uint256 balanceOfWant = balanceOfWant();
+
+        if (balanceOfWant > balanceInit) {
+            _profit = balanceOfWant.sub(balanceInit);
+        }
+
+        // if the vault is claiming repayment of debt
+        if (_debtOutstanding > 0) {
+            uint256 _amountFreed = 0;
+            (_amountFreed, _loss) = liquidatePosition(_debtOutstanding);
+            _debtPayment = Math.min(_debtOutstanding, _amountFreed);
```

Comment:
I think `_profit` might be double counted here since `liquidatePosition` takes into account the existing balance.
i.e  if `debtOutstanding = 10000` and `_profit = 100` 
-> `liquidatePosition(10000)`
-> `toWithdraw = 10000 - 100 = 9900`
-> `totalAssets = 10000`, `liquidatedAmount = 10000`
-> `(_amountFreed, _loss) = (10000,0)`
-> `_debtPayment = 10000`
now you have an extra 100 reported as profit but only 10000 of want free

Reply:
one solution would be to just withdraw the amount requested, without taking into account existing balance (`10000 - 100` part)

Reply:
another alternative w/o changing too much is to do `liquidatePosition(_debtOutstanding.add(balanceOfWant))`

Reply:
yup that works too

Reply:
@hsukev good catch, I am going to go with @orbxball recommendation.

Reply:
wouldn't it be better to use @orbxball 's but instead of balanceOfWant use _profit?

as in `liquidatePosition(_debtOutstanding.add(_profit))`